src/dashboard/setup.py

```
@@ -10,16 +10,35 @@
 """
 
 import os
+import subprocess
 
 from setuptools import setup
 
 
 def build_record_viewer():
     if os.path.exists("react_components/record_viewer"):
-        print("running npm i")
-        os.system("npm i --prefix react_components/record_viewer")
+        try:
+            print("running npm i")
+            subprocess.check_call(
+                [
+                    "npm",
+                    "i",
+                    "--prefix",
+                    "react_components/record_viewer",
+                ]
+            )
+        except FileNotFoundError:
+            print("`npm` is not installed. Skipping build")
```

We are not skipping build here? we need to raise the exception or exit here?